### PR TITLE
security group: fix component test test_multiple_nic_support.py failures 

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes component test failure with test_multiple_nic_support.py, in advanced zone with security groups.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

manually run the component test, all passed.

```
Create Vm with multiple NIC's ... === TestName: test_01_create_vm_with_multiple_nics | Status : SUCCESS ===
ok
Create VM with single NIC and then add additional NIC ... === TestName: test_02_add_nic_to_vm | Status : SUCCESS ===
ok
Add secondary IP's to the VM ... === TestName: test_03_add_ip_to_default_nic | Status : SUCCESS ===
ok
Add secondary IP's to remaining NIC's ... === TestName: test_04_add_ip_to_remaining_nics | Status : SUCCESS ===
ok
Stop and Start a VM with Multple NIC ... === TestName: test_05_stop_start_vm_with_multiple_nic | Status : SUCCESS ===
ok
Migrate a VM with Multple NIC ... === TestName: test_06_migrate_vm_with_multiple_nic | Status : SUCCESS ===
ok
Remove secondary IP from any NIC ... === TestName: test_07_remove_secondary_ip_from_nic | Status : SUCCESS ===
ok
Remove NIC from VM ... === TestName: test_08_remove_nic_from_vm | Status : SUCCESS ===
ok
Reboot a VM with Multple NIC ... === TestName: test_09_reboot_vm_with_multiple_nic | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 9 tests in 359.228s

OK

```

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
test with advanced zone is not needed actually.

@blueorangutan test

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
Trillian does not run SG tests, we can merge based on Wei's test results.

File: test/integration/component/test_multiple_nic_support.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def setUpClass(cls):
                     cls.virtual_machine1.default_network_id = nic.networkid
                     break
         except Exception as e:
-            cls.fail("Exception while deploying virtual machine: %s" % e)
+            cls.fail("Exception while deploying virtual machine: %s" % (e))
```
Comment (DaanHoogland) [inline test/integration/component/test_multiple_nic_support.py:227]:
```suggestion
            cls.fail(f"Exception while deploying virtual machine: {e}")
```
not necessary but nice ;)

File: test/integration/component/test_multiple_nic_support.py [hunk 2]
```diff
@@ -243,7 +243,7 @@ def setUpClass(cls):
                     cls.virtual_machine2.default_network_id = nic.networkid
                     break
         except Exception as e:
-            cls.fail("Exception while deploying virtual machine: %s" % e)
+            cls.fail("Exception while deploying virtual machine: %s" % (e))
```
Comment (DaanHoogland) [inline test/integration/component/test_multiple_nic_support.py:246]:
```suggestion
            cls.fail(f"Exception while deploying virtual machine: {e}")
```

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
just to clarify @weizhouapache , I suppose you created a advanced zone with security groups and ran the test-file against that?

kind of asking also to let you consider my non-vital suggestion. I'm fine to merge without.

File: test/integration/component/test_multiple_nic_support.py (hunk 1 above)
Comment (rhtyd) [reply test/integration/component/test_multiple_nic_support.py:227]: @DaanHoogland there's an `f"Ex...` is that a typo or something new in py3?

Comment (rhtyd) [reply test/integration/component/test_multiple_nic_support.py:227]: Nevermind, I googled - I need a bit a re-learning py3 https://realpython.com/python-f-strings/

File: test/integration/component/test_multiple_nic_support.py [hunk 3]
```diff
@@ -224,7 +224,7 @@ def setUpClass(cls):
                     cls.virtual_machine1.default_network_id = nic.networkid
                     break
         except Exception as e:
-            cls.fail("Exception while deploying virtual machine: %s" % e)
+            cls.fail(f"Exception while deploying virtual machine: {e}")
```
Comment (weizhouapache) [inline test/integration/component/test_multiple_nic_support.py:227]: @DaanHoogland 
a bit strange that this does not work in centos7/python2

```
======================================================================
ERROR: Failure: SyntaxError (invalid syntax (test_multiple_nic_support.py, line 227))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 132, in run
    self.beforeTest(result)
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 74, in beforeTest
    beforeTest(self.test)
  File "/usr/lib/python2.7/site-packages/nose/proxy.py", line 117, in beforeTest
    self.plugins.beforeTest(self.test)
  File "/usr/lib/python2.7/site-packages/nose/plugins/manager.py", line 99, in __call__
    return self.call(*arg, **kw)
  File "/usr/lib/python2.7/site-packages/nose/plugins/manager.py", line 167, in simple
    result = meth(*arg, **kw)
  File "/usr/lib/python2.7/site-packages/marvin/marvinPlugin.py", line 174, in beforeTest
    join([self.__identifier, self.__testName])
TypeError: sequence item 0: expected string, NoneType found
-------------------- >> begin captured stdout << ---------------------
=== TestName: Failure: | Status : EXCEPTION ===
```

Comment (DaanHoogland) [reply test/integration/component/test_multiple_nic_support.py:227]: not sure why, but the old way works (and still does in py3) no worries. I'll give it a look one of these days.

**Comment by DaanHoogland** [comment]
wait, this was merged with these suggestions? Do we need to revert?

**Comment by weizhouapache** [comment]
> wait, this was merged with these suggestions? Do we need to revert?

@DaanHoogland if it works with python3, no need to revert it. it is just component test which is rarely run.

**Comment by DaanHoogland** [comment]
ok, i didn't test it yet, but i will and amend if needed.

File: test/integration/component/test_multiple_nic_support.py (hunk 3 above)
Comment (DaanHoogland) [reply test/integration/component/test_multiple_nic_support.py:227]: ok, double checked:
```
>>> e = Exception("root cause")
>>> raise Exception(f"nested exception {e}")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: nested exception root cause
```
I'll fix if it turns out to be a problem for anyone ;)

**Pull request merged**